src/mito2/src/worker/handle_open.rs

```diff
@@ -71,18 +95,43 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             self.intermediate_manager.clone(),
         )
         .skip_wal_replay(request.skip_wal_replay)
-        .parse_options(request.options)?
         .cache(Some(self.cache_manager.clone()))
-        .open(&self.config, &self.wal)
-        .await?;
-
-        info!("Region {} is opened", region_id);
+        .parse_options(request.options)
+        {
+            Ok(opener) => opener,
+            Err(err) => {
+                sender.send(Err(err));
+                return;
+            }
+        };
 
-        REGION_COUNT.inc();
+        let regions = self.regions.clone();
+        let wal: crate::wal::Wal<S> = self.wal.clone();
+        let config = self.config.clone();
+        let opening_regions = self.opening_regions.clone();
+        opening_regions.insert_sender(region_id, sender);
+        common_runtime::spawn_bg(async move {
+            match opener.open(&config, &wal).await {
+                Ok(region) => {
+                    info!("Region {} is opened", region_id);
+                    REGION_COUNT.inc();
 
-        // Insert the MitoRegion into the RegionMap.
-        self.regions.insert_region(Arc::new(region));
+                    // Insert the Region into the RegionMap.
+                    regions.insert_region(Arc::new(region));
 
-        Ok(0)
+                    let senders = opening_regions.remove_sender(region_id);
```

Comment:
Be careful. It's easy to have concurrent issues if we check these two maps in a different order. Maybe we should insert the region into the `regions` map while holding the `opening_regions` lock.